process/smartContract/process.go

```diff
@@ -410,13 +410,13 @@ func (sc *scProcessor) createVMInput(tx data.TransactionHandler) (*vmcommon.VMIn
 	var err error
 	vmInput := &vmcommon.VMInput{}
 
-	vmInput.CallerAddr = tx.GetSndAddress()
+	vmInput.CallerAddr = tx.GetSndAddr()
 	vmInput.Arguments, err = sc.argsParser.GetArguments()
 	if err != nil {
 		return nil, err
 	}
 
-	vmInput.CallValue = tx.GetValue()
+	vmInput.CallValue = new(big.Int).Set(tx.GetValue().Get())
```

Comment:
any reason for creating a copy of big.Int tx.GetValue().Get() instead of setting it directly on vmInput.CallValue ?

Reply:
yes, if not doing this , we can just hope that the caller is not changing the pointed value